src/main/java/org/graylog/plugins/collector/collectors/rest/CollectorResource.java

```diff
@@ -118,8 +121,11 @@ public Response register(@ApiParam(name = "collectorId", value = "The collector
         final Collector collector = collectorService.fromRequest(collectorId, request, collectorVersion);
 
         collectorService.save(collector);
-
-        return Response.accepted().build();
+        final CollectorRegistrationResponse collectorRegistrationResponse = CollectorRegistrationResponse.create(
+                configSupplier.get().collectorUpdateInterval().getSeconds(),
+                configSupplier.get().collectorSendStatus()
```

Comment:
Calling `configSupplier.get()` twice does two database lookups. Maybe just assign the config to a variable.